core/ai.go

```diff
@@ -127,3 +132,102 @@ func ParseStepsFromModelID(modelID *string, defaultSteps float64) float64 {
 
 	return numInferenceSteps
 }
+
+// AddAICapabilities adds AI capabilities to the node.
+func (n *LivepeerNode) AddAICapabilities(caps *Capabilities) {
+	aiConstraints := caps.PerCapability()
+	if aiConstraints == nil {
+		return
+	}
+
+	n.Capabilities.mutex.Lock()
+	defer n.Capabilities.mutex.Unlock()
+	for aiCapability, aiConstraint := range aiConstraints {
+		_, capExists := n.Capabilities.constraints.perCapability[aiCapability]
+		if !capExists {
+			n.Capabilities.constraints.perCapability[aiCapability] = &CapabilityConstraints{
+				Models: make(ModelConstraints),
+			}
+		}
+
+		for modelId, modelConstraint := range aiConstraint.Models {
+			_, modelExists := n.Capabilities.constraints.perCapability[aiCapability].Models[modelId]
+			if modelExists {
+				n.Capabilities.constraints.perCapability[aiCapability].Models[modelId].Capacity += modelConstraint.Capacity
+			} else {
+				n.Capabilities.constraints.perCapability[aiCapability].Models[modelId] = &ModelConstraint{Warm: modelConstraint.Warm, Capacity: modelConstraint.Capacity}
+			}
+		}
+	}
+}
+
+// RemoveAICapabilities removes AI capabilities from the node.
+func (n *LivepeerNode) RemoveAICapabilities(caps *Capabilities) {
```

Comment:
This function should not exist

Reply:
Not following. Is this not similar to `RemoveCapacity`?

You are saying it would be better to squash this into `RemoveCapacity`? Without updating tests to cover this addition into `RemoveCapacity` function?

New function was added because transcoding does not use capabilities in this way right now and tried to avoid adding complication to a function used with remote transcoder connection that didn't need to be there.